src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,148 @@
+
+        User registeredUser = registrationService.register(validUser);
+        assertNotNull(registeredUser, "User should be registered");
+        assertEquals(validUser.getLogin(), registeredUser.getLogin());
+        assertEquals(validUser.getPassword(), registeredUser.getPassword());
+        assertEquals(validUser.getAge(), registeredUser.getAge());
+        assertEquals(1, Storage.people.size(), "User should be added to the storage");
+    }
+
+    @Test
+    public void register_loginTooShort_notOk() {
+        User user = new User();
+        user.setLogin("short");
+        user.setPassword("qwertyQ123");
+        user.setAge(20);
+
+        RegistrationException thrown = assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        assertEquals("Login must be at least 6 characters", thrown.getMessage());
+    }
+
+    @Test
+    public void register_passwordTooShort_notOk() {
+        User user = new User();
+        user.setLogin("validUser123");
+        user.setPassword("12345");
+        user.setAge(20);
+
+        RegistrationException thrown = assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        assertEquals("Password must be at least 6 characters", thrown.getMessage());
+    }
+
+    @Test
+    public void register_ageBelow18_notOk() {
+        User user = new User();
+        user.setLogin("validUser123");
+        user.setPassword("qwertyQ123");
+        user.setAge(17);
+
+        RegistrationException thrown = assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        assertEquals("User must be at least 18 years old", thrown.getMessage());
+    }
+
+    @Test
+    public void register_ageExactly18_ok() {
+        User user = new User();
+        user.setLogin("validUser123");
+        user.setPassword("qwertyQ123");
+        user.setAge(18);
+
+        User registeredUser = registrationService.register(user);
+        assertNotNull(registeredUser, "User should be registered");
+        assertEquals(user.getLogin(), registeredUser.getLogin());
+        assertEquals(user.getPassword(), registeredUser.getPassword());
+        assertEquals(user.getAge(), registeredUser.getAge());
+    }
+
+    @Test
+    public void register_duplicateLogin_notOk() {
+        User firstUser = new User();
+        firstUser.setLogin("validUser123");
+        firstUser.setPassword("qwertyQ123");
+        firstUser.setAge(20);
+        registrationService.register(firstUser);
+
+        User secondUser = new User();
+        secondUser.setLogin("validUser123");
+        secondUser.setPassword("password456");
+        secondUser.setAge(22);
+
+        RegistrationException thrown = assertThrows(RegistrationException.class,
+                () -> registrationService.register(secondUser));
+        assertEquals("User with this login already exists", thrown.getMessage());
```

Comment:
The test `register_duplicateLogin_notOk` assumes that the system should not allow duplicate logins. This requirement is not mentioned in the checklist. If it is part of the task, ensure that the `register` method in `RegistrationServiceImpl` checks for duplicate logins before adding a new user.